lib/snyk/package.go

```diff
@@ -29,6 +30,51 @@ import (
 const snykServer = "https://api.snyk.io/rest"
 const version = "2023-04-28"
 
+func purlToSnykURL(basePath string, purl packageurl.PackageURL, nsDelimiter string) string {
+	sb := strings.Builder{}
+	sb.WriteString(basePath)
+	if purl.Namespace != "" {
+		sb.WriteString(purl.Namespace)
+		sb.WriteString(nsDelimiter)
+	}
+	sb.WriteString(purl.Name)
+	return sb.String()
+}
+
+func purlToSnykAdvisor(purl packageurl.PackageURL) string {
+	return map[string]string{
+		packageurl.TypeNPM:    "npm-package",
+		packageurl.TypePyPi:   "python",
+		packageurl.TypeGolang: "golang",
+		packageurl.TypeDocker: "docker",
+	}[purl.Type]
+}
+
+func SnykAdvisorURL(pkgPath string, purl packageurl.PackageURL) string {
```

Comment:
I don't feel like it makes sense to export these functions in their current state. If I wanted to use them as a dependant of this package, I would have to call `purlToSnykAdvisor` in order to do so; which i cannot outside of this package. However, I think they are fine functions to be exported. How about changing them slightly:

```go
func SnykAdvisorURL(purl *packageurl.PackageURL) string { // only require purl as an input
    if purl == nil {
        return ""
    }

    ecosystem := snykAdvisorEcosystems[purl.type] // no need to have a separate function for this imho
    if ecosystem == "" { // check for ecosystem here
        return ""
    }

    // ...go on to construct URL
}
```